src/Native/System.Native/pal_networking.cpp

```diff
@@ -213,12 +216,33 @@ extern "C" int32_t IPv4StringToAddress(const uint8_t* address, uint8_t* buffer,
     return ConvertGetAddrInfoAndGetNameInfoErrorsToPal(result);
 }
 
+static void AppendScopeIfNecessary(uint8_t* string, int32_t stringLength, uint32_t scope)
```

Comment:
> Isn't this how libc and other platform libraries do it? And they're often giving back or filling in UTF8-encoded strings.

Yes, but this is mostly for reasons of consistency with legacy code. Such code often misuses `char*` for byte buffers as well. We have no reason here to repeat the mistakes of the past.